### PR TITLE
manifest: set the public and private zones for AWS 

### PR DESCRIPTION
**Pull request by abhinavdahiya**
The `DNS` object [2] was expanded to include the pointers to the public and private hosted zones to allow operators to create DNS records inthe correct zone in `[PR #202]`[4].

The public zone is fetched from the `basedomain` while since installer creates the private zone at later stage, tags are used to specify the private dns zone.

The ID returned by `get-hosted-zone` [1] is of the form `/hostedzone/<id>`. For example,

```console
$ AWS_PROFILE=openshift-dev aws route53 get-hosted-zone --id XXX
{
    "HostedZone": {
        "Id": "/hostedzone/XXX",
...
}
```

This creates the `DNS` [2] object like:
```yaml
apiVersion: config.openshift.io/v1
kind: DNS
metadata:
  name: cluster
spec:
  baseDomain: yyy.openshift.com
  privateZone:
    tags:
      Name: adahiya-1_int
      kubernetes.io/cluster/adahiya-1: owned
      openshiftClusterID: 25375c13-3c0a-4102-a8f9-7ecb60757c62
  publicZone:
    id: "/hostedzone/XXX"
status: {}
```

But you can actually use both `/hostedzone/XXX` or `XXX` to get a zone.

```console
$ AWS_PROFILE=openshift-dev aws route53 get-hosted-zone --id "/hostedzone/XXX"
{
    "HostedZone": {
        "Id": "/hostedzone/XXX",
...
}
```

The change trims the `zoneID` to not include the prefix when creating the `DNS` object as just the ID has cleaner semantics when stored in the object. The terraform-provider-aws also cleans the zone-id for better UX [3] by trimming the prefix `/hostedzone/`.

[1]: https://docs.aws.amazon.com/cli/latest/reference/route53/get-hosted-zone.html#synopsis
[2]: https://github.com/openshift/api/blob/8c839bc7ff62e38ad7656bf920e11e2664a44f6b/config/v1/types_dns.go#L11
[3]: https://github.com/terraform-providers/terraform-provider-aws/blob/75a9ebb7bfc7fd61b4454589155cc6b958ebebe4/aws/resource_aws_route53_zone.go#L461-L464
[4]: https://github.com/openshift/api/pull/202
/cc @ironcladlou

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/api/pull/202 merged. So this is ready for review.

/cc @wking @staebler @ironcladlou 

Need this so that ingress can make progress which is blocking https://github.com/openshift/installer/pull/1169

**Comment by ironcladlou** [comment]
This lgtm

File: pkg/asset/installconfig/aws/basedomain.go [hunk 1]
```diff
@@ -78,3 +79,30 @@ func GetBaseDomain() (string, error) {
 
 	return domain, nil
 }
+
+// GetPublicZone returns a public route53 zone that match the name.
```
Comment (staebler) [inline pkg/asset/installconfig/aws/basedomain.go]: s/match/matches/

File: pkg/asset/manifests/dns.go [hunk 2]
```diff
@@ -58,6 +71,23 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
...
+	case awstypes.Name:
+		zone, err := icaws.GetPublicZone(installConfig.Config.BaseDomain)
+		if err != nil {
+			return errors.Wrapf(err, "getting public zone for %s", installConfig.Config.BaseDomain)
+		}
+		config.Spec.PublicZone = &configv1.DNSZone{ID: strings.TrimPrefix(*zone.Id, "/hostedzone/")}
+		config.Spec.PrivateZone = &configv1.DNSZone{Tags: map[string]string{
+			"openshiftClusterID":                                                          clusterID.ClusterID,
+			fmt.Sprintf("kubernetes.io/cluster/%s", installConfig.Config.ObjectMeta.Name): "owned",
+			"Name": fmt.Sprintf("%s_int", installConfig.Config.ObjectMeta.Name),
```
Comment (staebler) [inline pkg/asset/manifests/dns.go]: Does the format need to be `%s-int` to match the name set for the api_internal load balancer?
https://github.com/openshift/installer/blob/17c0fa8d093193fb933b476fd70f7c627520bd6f/data/data/aws/vpc/master-elb.tf#L1-L2

Comment (abhinavdahiya) [reply pkg/asset/manifests/dns.go]: No this is to match https://github.com/openshift/installer/blob/17c0fa8d093193fb933b476fd70f7c627520bd6f/data/data/aws/main.tf#L116

Comment (staebler) [reply pkg/asset/manifests/dns.go]: Ah, yes. I couldn't find that last night for whatever reason.

**Comment by abhinavdahiya** [comment]
e2e flaking

/retest

File: pkg/asset/manifests/dns.go [hunk 3]
```diff
@@ -58,6 +71,23 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
...
+		}}
+	case libvirttypes.Name, openstacktypes.Name, nonetypes.Name:
+	default:
+		return errors.New("invalid Platform")
```
Comment (wking) [inline pkg/asset/manifests/dns.go]: Isn't our current plan to ignore invalid platforms in most places?  See [here][1] and [here][2].  I guess your current whitelist approach makes sense if we expect new platforms to require cluster-managed DNS configuration and that we feel a more permissive approach in this block would make it easier to forget to add it?

[1]: https://github.com/openshift/installer/pull/792#discussion_r253683216
[2]: https://github.com/openshift/installer/pull/1112/files#diff-6532666297c6115f7774f93ebab396c4R156

**Comment by wking** [comment]
[e2e-aws][1]:

```
Failing tests:

[Area:Networking] NetworkPolicy when using a plugin that implements NetworkPolicy should enforce policy based on Ports [Feature:OSNetworkPolicy] [Suite:openshift/conformance/parallel]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should serve the correct routes when scoped to a single namespace and label set [Suite:openshift/conformance/parallel/minimal]
[Conformance][Area:Networking][Feature:Router] The HAProxy router should support reencrypt to services backed by a serving certificate automatically [Suite:openshift/conformance/parallel/minimal]
[Conformance][templates] templateinstance readiness test  should report failed soon after an annotated objects has failed [Suite:openshift/conformance/parallel/minimal]
[Conformance][templates] templateinstance readiness test  should report ready soon after all annotated objects are ready [Suite:openshift/conformance/parallel/minimal]
[Feature:Builds][Conformance] build can reference a cluster service  with a build being created from new-build should be able to run a build that references a cluster service [Suite:openshift/conformance/parallel/minimal]
[Feature:DeploymentConfig] deploymentconfigs with multiple image change triggers [Conformance] should run a successful deployment with a trigger used by different containers [Suite:openshift/conformance/parallel/minimal]
[sig-api-machinery] CustomResourceDefinition Watch CustomResourceDefinition Watch watch on custom resource definition objects [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-api-machinery] Servers with support for Table transformation should return chunks of table results for list calls [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-apps] Deployment deployment should support proportional scaling [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should provide basic identity [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl Port forwarding [k8s.io] With a server listening on localhost [k8s.io] that expects a client request should support a client that connects, sends DATA, and disconnects [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl Port forwarding [k8s.io] With a server listening on localhost should support forwarding over websockets [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Kubectl apply should reuse port when apply to an existing SVC [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] HostPath should support subPath [NodeConformance] [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: cinder] [Testpattern: Dynamic PV (default fs)] subPath should support readOnly file specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: emptydir] [Testpattern: Inline-volume (default fs)] subPath should support existing directory [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: emptydir] [Testpattern: Pre-provisioned PV (default fs)] subPath should support existing directory [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Dynamic PV (default fs)] subPath should support readOnly file specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Inline-volume (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: hostPathSymlink] [Testpattern: Inline-volume (default fs)] subPath should support existing single file [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (ext3)] volumes should be mountable [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: vSphere] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
```

/retest

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1233/pull-ci-openshift-installer-master-e2e-aws/3668/build-log.txt

**Pull request merged**